mailpoet/lib/Automation/Engine/Control/StepHandler.php

```diff
@@ -113,10 +113,11 @@ private function handleStep(array $args): void {
       throw new InvalidStateException();
     }
 
+    $nextStep = $step->getNextSteps()[0] ?? null;
     $nextStepArgs = [
       [
         'workflow_run_id' => $workflowRunId,
-        'step_id' => $step->getNextStepId(),
+        'step_id' => $nextStep ? $nextStep->getId() : null,
```

Comment:
It might be nice to update the check on line 130 to look at `$nextStep` instead of digging into `$nextStepArgs`.

Reply:
True! Fixed https://github.com/mailpoet/mailpoet/pull/4359/commits/a70b2606e0463890635a1c776f5f7c65aec72f75, thanks.